src/basic/path-util.c

```diff
@@ -518,7 +517,7 @@ char* path_join_internal(const char *first, ...) {
         if (!joined)
                 return NULL;
 
-        if (first[0] != 0) {
+        if (!isempty(first)) {
```

Comment:
urks, this is magic.

Here's another idea. strjoin_many() is a macro anyway, which means there's no benefit in using NULL as end marker for the list anyway (as we don't need to be nice to gcc's sentinel attribute). Hence, maybe instead of ending the argument list with `NULL` we could just end it in `(char*) -1`. That would mean we could always consider NULL to be equivalent to "" as we don't need to overload it with two meanings anymore!